lib/xandra.ex

```diff
@@ -1215,17 +1214,15 @@ defmodule Xandra do
     end
   end
 
-  defp execute_with_retrying(conn, query, params, options) do
+  defp execute_query(conn, query, params, options) do
     {xandra_opts, db_conn_opts} = Keyword.split(options, @execute_opts_keys)
     xandra_opts = NimbleOptions.validate!(xandra_opts, @execute_opts_schema)
     options = xandra_opts ++ db_conn_opts
 
-    RetryStrategy.run_with_retrying(options, fn ->
-      execute_without_retrying(conn, query, params, options)
-    end)
```

Comment:
Mh, am I wrong or we are removing retrying altogether from non-cluster connections? If so, then we shouldn't do that 🙃 

Let's also avoid renaming functions to `do_` prefixes, since it's a long-gone convention in Elixir.

Reply:
Ah I thought that when elevating retries to cluster level, we both meant that we don't want `Xandra` level retries anymore. Is the reason why we don't let the client implement a helper function but to have a retry strategy is to not checkout the `Xandra` connection pool multiple times? Because that would be the only logical reason to have a retry logic, since everything else can be done with more context with a helper function on the client side.

Otherwise, if we want to have the retry logic on Xandra level as well, maybe we want to have a different callback than `Xandra.RetryStrategy`? Since the `retry/3` function is expected to return a `Xandra` connection pool, we can't really have the same callback for `Xandra.Cluster` and `Xandra`.

Reply:
> Is the reason why we don't let the client implement a helper function but to have a retry strategy is to not checkout the Xandra connection pool multiple times? Because that would be the only logical reason to have a retry logic

I disagree, and I think calling this "the only logical reason" can come off a bit arrogant 😉  Logical reasons to have retry logics are:

  1. We would introduce a breaking change by removing them. This might not be a good reason for having introduced them, but alas, it's what we have!
  2. It's easier to reuse retry logic modules provided by Xandra or other people. The intention initially was to provide other built-in retry strategies, such as retrying with lower consistency requirements and stuff like that. That never really happened, but still 🙃 

> Otherwise, if we want to have the retry logic on Xandra level as well, maybe we want to have a different callback than `Xandra.RetryStrategy`?

As I said in other comments, we can use the options for all of this. We can inject the node to use in the options, and use that in `Xandra.Cluster.<...>` functions if present.

Reply:
Sorry, didn't mean to be that harsh :) I implemented the requested changes now, with the API still the same.